akka-remote/src/test/scala/akka/remote/artery/tcp/TlsTcpSpec.scala

```diff
@@ -53,6 +53,27 @@ class TlsTcpWithCrappyRSAWithMD5OnlyHereToMakeSureThingsWorkSpec
     }
     """))
 
+class TlsTcpWithRefreshingProviderSpec
+    extends TlsTcpSpec(ConfigFactory.parseString(s"""
+    akka.remote.artery.ssl {
+      ssl-engine-provider = "akka.remote.artery.tcp.RefreshingConfigSSLEngineProvider"
+      refreshing-ssl-engine {
+        key-file = ${getClass.getClassLoader.getResource("ssl/pem/pkcs1.pem").getPath}
+        # TODO ideally we could use different certificates for the server and client roles
+        cert-file = ${getClass.getClassLoader.getResource("ssl/pem/certificate.pem").getPath}
+        ca-cert-file = ${getClass.getClassLoader.getResource("ssl/pem/certificate.pem").getPath}
+        ssl-context-cache-ttl = 1d
+
+        # TODO duplicated from config-ssl-engine - let's see if we can consolidate that?
```

Comment:
+1

`protocol`, `enabled-algorithms` and `random-number-generator` may be extracted and consolidated but not `hostname-verification` and `require-mutual-authentication`.

I think that's what you mentioned but wanted to disambiguate it. :-)

Reply:
Well, I would *like* to also consolidate `hostname-verification` and `require-mutual-authentication`, but perhaps that's not feasible and indeed we won't. Let's see how far we can take this ;).